zathura/synctex.c

```diff
@@ -16,22 +17,74 @@
 #include "adjustment.h"
 
 #ifdef WITH_SYNCTEX
+synctex_scanner_p scanner = NULL; // the last scanner object
+time_t last_modification_time = 0; // the last modification time of the synctex file
+char* last_pdf_file_name = NULL; // the last output file name
```

Comment:
This state would need to go to the `zathura_t` struct. When the document is closed and reopened, also the scanners can be reopened.

Reply:
What do you mean by "the scanners can be reopened"?

I don't think reloading the synctex file early is a good idea consider that for a large document it can take up to a few seconds, which would make zathura look sluggish.

An alternative is to load it in a separate thread — that way CPU is consumed but otherwise nothing bad happens. What do you think?

Reply:
I meant that when the document is closed, the scanner is closed as well. When the file is reopened and on the next synctex operation, the scanner is then also reloaded.